tests/unit/python/foglamp/services/core/service_registry/test_monitor.py

```diff
@@ -0,0 +1,109 @@
+__version__ = "${VERSION}"
+
+
+@pytest.allure.feature("unit")
+@pytest.allure.story("core", "monitor")
+class TestMonitor:
+    @pytest.fixture
+    def reset_service_registry(self):
+        del ServiceRegistry._registry[:]
+        yield
+        del ServiceRegistry._registry[:]
+
+    @pytest.mark.asyncio
+    async def test__monitor_good_uptime(self, reset_service_registry):
+        async def async_mock(return_value):
+            return return_value
+        # used to mock client session context manager
+
+        class AsyncSessionContextManagerMock(MagicMock):
+            def __init__(self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+
+            async def __aenter__(self):
+                client_response_mock = MagicMock(spec=aiohttp.ClientResponse)
+                # mock response (good)
+                client_response_mock.text.side_effect = [
+                    async_mock('{"uptime": "bla"}')]
+                return client_response_mock
+
+            async def __aexit__(self, *args):
+                return None
+        # as monitor loop is as infinite loop, this exception is thrown when we need to exit the loop
+
+        class TestMonitorException(Exception):
+            pass
+        # register a service
+        s_id_1 = ServiceRegistry.register(
+            'sname1', 'Storage', 'saddress1', 1, 1,  'protocol1')
+        monitor = Monitor()
+        monitor._sleep_interval = Monitor._DEFAULT_SLEEP_INTERVAL
+
+        # throw the TestMonitorException when sleep is called (end of infinite loop)
+        with patch.object(Monitor, '_sleep', side_effect=TestMonitorException()):
+            with patch.object(aiohttp.ClientSession, 'get', return_value=AsyncSessionContextManagerMock()):
+                with pytest.raises(TestMonitorException) as excinfo:
+                    await monitor._monitor_loop()
+        # service is good, so it should remain in the service registry
+        assert len(ServiceRegistry.get(idx=s_id_1)) is 1
+        assert ServiceRegistry.get(idx=s_id_1)[0]._status is 1
+
+    @pytest.mark.asyncio
+    async def test__monitor_exceed_attempts(self, reset_service_registry):
+        async def async_mock(return_value):
+            return return_value
+        # used to mock client session context manager
+
+        class AsyncSessionContextManagerMock(MagicMock):
+            def __init__(self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+
+            async def __aenter__(self):
+                # mock response (error- exception)
+                raise Exception()
+
+            async def __aexit__(self, *args):
+                return None
+        # as monitor loop is as infinite loop, this exception is thrown when we need to exit the loop
+
+        class TestMonitorException(Exception):
+            pass
+        # register a service
+        s_id_1 = ServiceRegistry.register(
+            'sname1', 'Storage', 'saddress1', 1, 1,  'protocol1')
```

Comment:
I see most of the code is repeating for both test scenarios, just asking. refactor?